### ISSUE TITLE
Code analysis warnings not treated as errors despite property

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#49882]
**Version Used**: 
dotnet SDK 5.0.100

**Steps to Reproduce**:

On a netstandard2.0 project, introduce a code analysis warning. Build with properties to treat warnings as errors.,

Minimal repro project: [cabug.zip](https://github.com/dotnet/roslyn/files/5669150/cabug.zip)

**Expected Behavior**:

Build fails with error.
I expect just `TreatWarningsAsErrors: true` should be sufficient, but just to be sure I also set `CodeAnalysisTreatWarningsAsErrors: true` based on [this documentation](https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props#codeanalysistreatwarningsaserrors).

**Actual Behavior**:

Build succeeds after emitting a warning.

**Comment by mavasani** [comment]
Duplicate of https://github.com/dotnet/roslyn/issues/43051. Verified the issue is fixed if project file explicitly references a newer compiler toolset:

```xml
  <ItemGroup>
    <PackageReference Include="Microsoft.Net.Compilers.Toolset" Version="3.9.0-2.final">
      <PrivateAssets>all</PrivateAssets>
      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
    </PackageReference>
  </ItemGroup>
```
